src/ts/reader.ts

```diff
@@ -1,8 +1,10 @@
 import { BigNumberish, Contract } from "ethers";
 
-type ReaderMethods = "getSolverAt" | "numSolvers";
+type ReaderMethods = "getSolverAt" | "numSolvers" | "filledAmountsForOrders";
```

Comment:
I think this should have separate `ReaderMethods` and `ReaderParameter` types, since the `SettlementStorageReader` contract doesn't have `getSolverAt` and `numSolver` methods.

Reply:
Yes, so I initially intended to do this, but got a bit confused at how to go about generalizing `readStorage` method to accept two different types of methods and arguments. I will take a stab at it, could be as easy as using a `|` here instead